### DISCUSSION TITLE
Something about the Competition Data

### DISCUSSION THREAD

**Discussion by yyLeaves** [Kaggle-Competitions-Code/k12-curriculum-recommendations#3] [General]
![31675343851_ pic](https://user-images.githubusercontent.com/76979429/216582988-e8ed6878-9cd3-49d2-b140-f0d8b31dfd3d.jpg)

|file|Train|Test|
|-|-|-|
|topics.csv|77k (61.5k Has Content + 15.5k None Leaf)|10k additional topics (none present in the training set)|
|content.csv|154k (appear in at least 1 topic)|a large number of additional content items only correlated to test set topics + **original content data in training set** (“superset of the public content.csv rows”)|
|correlation|61.5k correlations.csv|sample_submission.csv|

**NOTES:**

1. content测试集中包含训练集的数据(是训练集的superset)
2. topics测试集为全新内容与训练集无重复。(应该同训练集一样有层级结构)
3. sample_submission包含要预测的topic_id
4. 少部分非叶子节点也有content(2.8k)
5. 一个channel对应一个根节点

<details>

1. [Where is the test data?](https://www.kaggle.com/competitions/learning-equality-curriculum-recommendations/discussion/372425#2067617)
   > In the test environment, the files content.csv and topics.csv will be replaced by the testing variants of these files, which will include additional withheld content items and topics.
   > 
   > The rows in sample_submission.csv determine which specific withheld topics you are expected to make predictions for. The public version of this file contains only a few rows, to demonstrate the format; in the testing environment, it will be replaced with a larger file, and you will need to fill in the content_ids column. Note that in the testing environment, correlations.csv will not change (as that would give away the answers).

2. [What is the full data set?](https://www.kaggle.com/competitions/learning-equality-curriculum-recommendations/data)
   > The full test set includes an additional 10,000 topics (none present in the training set) and a large number of additional content items. The additional content items are only correlated to test set topics.

3. [Will contents in the train data appear in the test data?](https://www.kaggle.com/competitions/learning-equality-curriculum-recommendations/discussion/372459#2068598)
   > In a submission run, test content.csv may have some ids from train content.csv, and test topics.csv are correlated with a something of train contents.
   >> That's correct. The topics you're predicting for will all be new and previously unseen, but some of the content will already have been seen during training. If you're optimizing for the efficiency track, and are using an embedding-based approach, you may wish to include a cache of the known content embeddings in your submission, to avoid needing to recompute them during the test run.

4. [Can I recommend similarities in a different language](https://www.kaggle.com/competitions/learning-equality-curriculum-recommendations/discussion/372549#2067436)
   > The goal of the competition is focused in K-12, so having similar languages as Norwegian and Danish is not appropiate. I can't think Norwergian schools use text books in Danish.
   >
   >For cases as Switzerland, it all will depend on what the curriculum states: if it says it's French content, it's not appropiate for an Italian spoken kid. You must bear in mind always the final consumer of the content will be kids, and school curriculums should always reference the language to be used in the content.

6. [Same title contents don't necessarily relate to the same topic](https://www.kaggle.com/competitions/learning-equality-curriculum-recommendations/discussion/374621#2081813)
   > Content items that are similar like that may come about from different versions of the same content being imported into the platform at different times or in different ways. It's likely an edge case, but worth noting. Recommending both of those content items for a topic like that shouldn't be penalized too much given we're using the F2 score that prioritizes recall.

7. [Channel & Content](https://www.kaggle.com/competitions/learning-equality-curriculum-recommendations/discussion/377381#2094957)
   > 1) It's a combination of both. (entirely new channels + topics that belong to only channels seen in the train set)
   >
   > 2) Topic trees in topics.csv should all have referential integrity (parent references should always take you up to the root, with level 0). Only subtrees are pruned.
   >
   >3) Yes, any content item in the version of content.csv in the test dataset is a candidate for prediction. This is a **superset** of the public content.csv rows.
</details>

**Comment by EeyoreLee** [comment]
test的数据应该是77k+10k的，test的topics和content都是train数据的超集，应该是为了submit时可以训练，预测只预测submit时submission中的topic

**Reply by EeyoreLee** [reply]
test topics数据集不会包含category为source的情况。

**Comment by EeyoreLee** [comment]
非常关键
```
Question about test set channels
I had similar questions to RoMoFlow in a previous post about the tree structure in the test set.

1) Does the test set consist of entirely new channels? Conversely, does it consist of topics that belong to only channels seen in the train set? Or is it a combination of both?
2) Will there be any instance of a topic that is not level 0 and has no parent? In other words, will there be incomplete topic trees?
3) Can the content in the training set be included as curriculum for the test topics?

Any pointers are much appreciated - thanks!
```
```
1) It's a combination of both.

2) Topic trees in topics.csv should all have referential integrity (parent references should always take you up to the root, with level 0). Only subtrees are pruned.

3) Yes, any content item in the version of content.csv in the test dataset is a candidate for prediction. This is a superset of the public content.csv rows.
```
